### PR TITLE
Adding lift from graph to Vietoris-Rips complex.

### PR DESCRIPTION
**Pull request by rballeba**
This commit creates a method to convert a graph into a Vietoris-Rips clique complex. It mimics the behaviour of the previous method `graph_2_clique_complex` but only adds cliques whose nodes have pairwise distances less than a radius parameter of the function. Unit tests added.

### CODE REVIEW

File: test/transform/test_graph_to_simplicial_complex.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 from toponetx.transform.graph_to_simplicial_complex import (
     graph_2_clique_complex,
     graph_2_neighbor_complex,
+    weighted_graph_to_Vietoris_Rips_complex
 )
```
Comment (mhajij) [inline test/transform/test_graph_to_simplicial_complex.py:12]: change weighted_graph_to_Vietoris_Rips_complex to weighted_graph_2_Vietoris_Rips_complex

to be consistent with other naming

Comment (rballeba) [reply test/transform/test_graph_to_simplicial_complex.py:12]: Changed in the new commit.

**Comment by ffl096** [comment]
Docs failure is unrelated (again #113), but lint failures should be fixed before merging.

File: toponetx/transform/graph_to_simplicial_complex.py [hunk 2]
```diff
@@ -60,3 +62,40 @@ def graph_2_clique_complex(G, max_dim=None):
 
     lst = filter(lambda face: len(face) <= max_dim, nx.enumerate_all_cliques(G))
     return SimplicialComplex(list(lst))
+
+
+def weighted_graph_to_Vietoris_Rips_complex(G, r, max_dim=None):
```
Comment (ninamiolane) [inline toponetx/transform/graph_to_simplicial_complex.py]: python style prevents uppercase letters in function names: 
```weighted_graph_to_vietoris_rips_complex```

File: toponetx/transform/graph_to_simplicial_complex.py [hunk 3]
```diff
@@ -60,3 +62,40 @@ def graph_2_clique_complex(G, max_dim=None):
...
+    """Get the Vietoris-Rips complex of radius r of a weighted undirected graph. The Vietoris-Rips complex of radius
```
Comment (ninamiolane) [inline toponetx/transform/graph_to_simplicial_complex.py:68]: Docstring one-linear should fit on one line. Then the description comes.

See anatomy of a docstring:
https://pyt-team.github.io/toponetx/contributing/index.html#docstring-examples

File: toponetx/transform/graph_to_simplicial_complex.py [hunk 4]
```diff
@@ -60,3 +62,40 @@ def graph_2_clique_complex(G, max_dim=None):
...
+        the output clique complex.
+        The default is None indicate max dimension.
+
+    Returns
+    -------
+    SimplicialComplex
+        The Vietoris-Rips simplicial complex of dimension max_dim of the graph G.
+        """
+
+    def is_in_VR_complex(clique):
```
Comment (ninamiolane) [inline toponetx/transform/graph_to_simplicial_complex.py:92]: Ditto

File: toponetx/transform/graph_to_simplicial_complex.py [hunk 5]
```diff
@@ -60,3 +62,40 @@ def graph_2_clique_complex(G, max_dim=None):
...
+        # Check if all edge weights are less or equal than r
```
Comment (ninamiolane) [inline toponetx/transform/graph_to_simplicial_complex.py:93]: No unnecessary comments, these are hard to maintain. Put this in a docstring.

File: test/transform/test_graph_to_simplicial_complex.py [hunk 6]
```diff
@@ -52,6 +53,100 @@ def test_graph_2_clique_complex(self):
...
+            # We add an edge between the two cliques with weight 3.0
+            G.add_edge(3, 4, weight=3.0)
+            # We add a new vertex 7 to the graph that forms a 3 clique with the previous
+            # vertices 4, 5, and 6 if we take r > 4.0
+            G.add_edge(4, 7, weight=4.0)
+            G.add_edge(5, 7, weight=4.0)
+            G.add_edge(6, 7, weight=4.0)
+            return G
+
+        def generate_expected_simplices_for_Vietoris_Rips_complex(r):
```
Comment (ninamiolane) [inline test/transform/test_graph_to_simplicial_complex.py:89]: no uppercase letters in function names in python.

File: test/transform/test_graph_to_simplicial_complex.py [hunk 7]
```diff
@@ -52,6 +53,100 @@ def test_graph_2_clique_complex(self):
...
+            """Generates a pair of lists of tuples containing the expected and non-expected simplices of the
```
Comment (ninamiolane) [inline test/transform/test_graph_to_simplicial_complex.py:90]: Docstring need to respect formatting + convention given in the contributing guidelines on the website. E.g. Generates --> Generate + other minor issues.

**Review by ninamiolane — APPROVED** [review]
Very nice! Some minor comments on style, and the Lint workflow needs to pass.

File: test/transform/test_graph_to_simplicial_complex.py (hunk 7 above)
Comment (rballeba) [reply test/transform/test_graph_to_simplicial_complex.py]: All the comments applied! Thank you @ninamiolane!

File: test/transform/test_graph_2_simplicial_complex.py [hunk 8]
```diff
@@ -0,0 +1,169 @@
+"""Test graph to simplicial complex transformation."""
```
Comment (ninamiolane) [inline test/transform/test_graph_2_simplicial_complex.py:1]: The name of this file should be test_[name of file being tested]

i.e test_graph_to_simplicial_complex

File: test/transform/test_graph_2_simplicial_complex.py [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+import unittest
```
Comment (ninamiolane) [inline test/transform/test_graph_2_simplicial_complex.py:3]: remove unittest dependency, we're trying to get rid of this and use pytest only.

File: test/transform/test_graph_2_simplicial_complex.py [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+import networkx as nx
+
+from toponetx.transform.graph_to_simplicial_complex import (
+    graph_2_clique_complex,
+    graph_2_neighbor_complex,
+    weighted_graph_2_vietoris_rips_complex,
+)
+
+
+class TestGraphToSimplicialComplex(unittest.TestCase):
```
Comment (ninamiolane) [inline test/transform/test_graph_2_simplicial_complex.py:14]:
```suggestion
class TestGraphToSimplicialComplex:
```

File: test/transform/test_graph_2_simplicial_complex.py (hunk 9 above)
Comment (ninamiolane) [inline test/transform/test_graph_2_simplicial_complex.py:3]:
```suggestion
```

File: test/transform/test_graph_2_simplicial_complex.py (hunk 8 above)
Comment (ninamiolane) [reply test/transform/test_graph_2_simplicial_complex.py:1]: I saw that the conversion functions themselves use "_2_". Could you rename this to "_to_" everywhere, for consistency, as well?

Many thanks.

**Review by ninamiolane — CHANGES_REQUESTED** [review]
Thank you! very minor comment.

File: test/transform/test_graph_2_simplicial_complex.py (hunk 9 above)
Comment (rballeba) [reply test/transform/test_graph_2_simplicial_complex.py]: Done!

Comment (rballeba) [reply test/transform/test_graph_2_simplicial_complex.py]: Done!

**Review by mhajij — APPROVED** [review]

File: toponetx/transform/graph_to_simplicial_complex.py (hunk 5 above)
Comment (mhajij) [reply toponetx/transform/graph_to_simplicial_complex.py]: this has not been addressed

Comment (rballeba) [reply toponetx/transform/graph_to_simplicial_complex.py]: Hi @mhajij! I removed this comment in the latest commit, but I forgot to answer Nina. Is it right now?

**Comment by ffl096** [comment]
@ninamiolane Your requested changes should have been addressed sufficiently. Can you please check and update your review accordingly? I cannot merge a pull request with a pending review by an owner 😅

**Comment by mhajij** [comment]
@rballeba  we can also close it and start a new one given the the conflict, up to you how you want to resolve it

**Comment by rballeba** [comment]
@mhajij As far as I know, the changes were solved. I will create a pull request again!

**Pull request closed without merging**